### PR TITLE
Faster and better caching, pagination and more

### PR DESCRIPTION
**Pull request by davidpofo**
Replaced the function-based view components_selected with class-based listview SelectedComponentsList and ProjectList. The latter also has pagination and is ordered by date created. Some caching and optimizations.

### CODE REVIEW

**Comment by gregelin** [comment]
@davidpofo Can you check my updates to `CHANGELOG.md` for completeness, please?

File: siteapp/settings.py [hunk 1]
```diff
@@ -136,6 +136,10 @@ def make_secret_key():
 	'simple_history.middleware.HistoryRequestMiddleware',
 	'pyinstrument.middleware.ProfilerMiddleware',
 ]
+CACHE_MIDDLEWARE_ALIAS='default'
+CACHE_MIDDLEWARE_SECONDS=604800
```
Comment (gregelin) [inline siteapp/settings.py:140]: @davidpofo Can you add some comments regarding what this cache does?

File: siteapp/settings_application.py [hunk 2]
```diff
@@ -1,10 +1,9 @@
-# nothing here
-
 import re
+import sys
 from .settings import *
 
 INSTALLED_APPS += [
-    #'debug_toolbar',
+    'debug_toolbar',
```
Comment (gregelin) [inline siteapp/settings_application.py:6]: @davidpofo Do we want Debig toolbar running all the time?

**Review by gregelin — CHANGES_REQUESTED** [review]

File: siteapp/settings_application.py (hunk 2 above)
Comment (davidpofo) [reply siteapp/settings_application.py:6]: Added a new enable_toolbar field in environment.json/settings to turn this on/off.

File: siteapp/settings.py (hunk 1 above)
Comment (davidpofo) [reply siteapp/settings.py:140]: I just removed as this was an attempt at site-wide caching that is more trouble than it is worth

Comment (gregelin) [reply siteapp/settings.py:140]: I have an idea on improving performance. Let's discuss after the rATO standup?

File: siteapp/settings_application.py (hunk 2 above)
Comment (gregelin) [reply siteapp/settings_application.py:6]: Please remember to `enable_toolbar` param to documentation: https://govready-q.readthedocs.io/en/latest/installation-and-setup/configuration-with-environment-variables.html

**Comment by davidpofo** [comment]
PR party 2/11. Future enhancements. @kerryrm not sands-serif it is san-serif.

**Pull request merged**